plugins/techdocs/src/reader/components/TechDocsReaderPageContent/TechDocsReaderPageContent.tsx

```diff
@@ -102,11 +103,12 @@ export const TechDocsReaderPageContent = withTechDocsReaderProvider(
     const handleAppend = useCallback(
       (newShadowRoot: ShadowRoot) => {
         setShadowRoot(newShadowRoot);
+        incShadowRootVersion();
```

Comment:
My best guess is that because the shadowRoot variable points to the DOM elements when the underlying DOM changes and the setShadowRoot call compares the old and new values to see if they're different they're the same so no side-effects get triggered... So we use the counter as a proxy to trigger these.